src/views/Graph.vue

```diff
@@ -622,14 +623,15 @@ export default {
         await this.$nextTick()
       }
     },
+    /** re-layout the graph after any new nodes have been rendered */
     async layout (nodes, edges, nodeDimensions) {
-      // re-layout the graph after any new nodes have been rendered
+      const graphviz = await Graphviz.load()
 
       // generate the GraphViz dot code
       const dotCode = this.getDotCode(nodeDimensions, nodes, edges)
 
       // run the layout algorithm
-      const jsonString = await graphviz.layout(dotCode, 'json')
+      const jsonString = graphviz.layout(dotCode, 'json')
```

Comment:
Does this make `layout` blocking?

Reply:
> The await in 1.x meant this was blocking anyway

???

`await` waits for a promise to be completed which should be async i.e. non-blocking right?

I think because the layout is happening in WASM the execution happens outside of the JS thread, but I've no idea how that works with blocking calls.

Reply:
It looks like HPCC is using the "web worker" pattern which should allow it to run in another thread:

https://github.com/hpcc-systems/hpcc-js-wasm/blob/d757dba9f6511bc81ad264b573de8dc6400492ea/index-worker.js

Reply:
> `await` waits for a promise to be completed which should be async i.e. non-blocking right?

Oh you mean blocking in the overall stack as opposed to this context

Reply:
Blocking means prevents the program from proceeding e.g in Python land:

```python
async def foo():
    blocking()
    await non_blocking()
```

I don't understand the JS implementation details but I think it's vaguely equivalent to Python's cooperative coroutine  approach:

* `await` suspends the coroutine and yields control back to the event-loop to pick another coroutine to run/resume.
* Returning from an `async` function lists the calling coroutine as ready to resume when the event-loop gets around to it.
* In JS `Promise` objects are used under the hood as proxies for the async execution.

In this code my worry is that there's no `await` on the `layout` which suggests that this operation is blocking. If so that would block the entire application preventing any UI updates whilst the layout is running. If that's true it's really bad. But I don't understand how webworkers interface with JS or the details of the JS implementation at all so I could be totally wrong.

Reply:
Bump to https://github.com/cylc/cylc-ui/issues/1298